aten/src/ATen/native/mkldnn/MKLDNNConversions.cpp

```diff
@@ -0,0 +1,46 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Config.h>
+#include "MKLDNNCommon.h"
+
+namespace at { namespace native {
+
+#if AT_MKLDNN_ENABLED()
+
+Tensor mkldnn_to_dense(const Tensor& mkldnn_tensor) {
+  ideep::tensor& stensor = itensor_from_mkldnn(mkldnn_tensor);
+  auto dims = stensor.get_dims();
+  // NOTE: int32_t dims from ideep::tensor but sizes needs int64_t
+  Tensor cpu_tensor = at::zeros(
+    std::vector<int64_t>(dims.begin(), dims.end()),
+    mkldnn_tensor.options().layout(c10::kStrided));
+  stensor.reorder_to(cpu_tensor.template data<float>());
+  return cpu_tensor;
+}
+
+Tensor dense_to_mkldnn(const Tensor& cpu_tensor) {
+  AT_ASSERTM(cpu_tensor.type_id() == CPUTensorId(),
+             "dense_to_mkldnn expects dense CPU tensor input");
+  AT_ASSERTM(cpu_tensor.scalar_type() == ScalarType::Float,
+             "dense_to_mkldnn expects float tensor input");
+  Tensor mkldnn_tensor = new_with_sizes_mkldnn(cpu_tensor.sizes(), cpu_tensor.options());
```

Comment:
@gchanan That's true, will add an additional check.

Reply:
@gchanan Hmm, I'd prefer to make `cpu_tensor` contiguous here by `cpu_tensor.contiguous()` before converting to `mkldnn_tensor`. This can free users from worrying about contiguous/non-contiguous when calling `to_mkldnn`. Moreover, some grads are non-contiguous (e.g. those from sum backward) so converting them back to mkldnn tensor also needs an extra `contiguous` call.

Reply:
sure, that's fine.  It would be even better if the mkldnn conversion could handle non-contiguous tensors without having to convert to contiguous, but this should work.

Reply:
> sure, that's fine. It would be even better if the mkldnn conversion could handle non-contiguous tensors without having to convert to contiguous, but this should work.

@gchanan Exactly. We will consider that. Thanks.